src/Analyzers/Analyzers/Serializability/ViewModelSerializabilityAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+            uninstantiableTypeTitle,
+            uninstantiableTypeMessage,
+            DiagnosticCategory.Serializability,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            uninstantiableTypeDescription);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(
+                UseSerializablePropertiesRule,
+                DoNotUseFieldsRule,
+                DoNotUseUninstantiablePropertiesRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.EnableConcurrentExecution();
+            context.RegisterSemanticModelAction(AnalyzeSyntaxTree);
+        }
+
+        private static void AnalyzeSyntaxTree(SemanticModelAnalysisContext context)
+        {
+            var semanticModel = context.SemanticModel;
+            var syntaxTree = context.SemanticModel.SyntaxTree;
+            var viewModelInterface = semanticModel.Compilation.GetTypeByMetadataName(dotvvmViewModelInterfaceMetadataName);
+            var bindAttribute = semanticModel.Compilation.GetTypeByMetadataName(dotvvmBindAttributeMetadataName);
+
+            // Check all classes
+            foreach (var classDeclaration in syntaxTree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>())
+            {
+                // Filter out non-ViewModels
+                var classInfo = semanticModel.GetDeclaredSymbol(classDeclaration);
+                if (!classInfo.AllInterfaces.Any(symbol => symbol.OriginalDefinition.Equals(viewModelInterface)))
+                    continue;
+
+                AnalyzeViewModelProperties(classDeclaration, bindAttribute, context);
+                AnalyzeViewModelFields(classDeclaration, context);
+            }
+        }
+
+        /// <summary>
+        /// Notify user about unserializable properties and unsupported properties
+        /// </summary>
+        /// <param name="viewModel">ViewModel class declaration</param>
+        /// <param name="context">Semantic context</param>
+        private static void AnalyzeViewModelProperties(ClassDeclarationSyntax viewModel, INamedTypeSymbol bindAttribute, SemanticModelAnalysisContext context)
+        {
+            var semanticModel = context.SemanticModel;
+            foreach (var property in viewModel.DescendantNodes().OfType<PropertyDeclarationSyntax>()
+                .Where(f => f.Modifiers.Any(m => m.Kind() == SyntaxKind.PublicKeyword)))
+            {
+                if (semanticModel.GetDeclaredSymbol(property) is not IPropertySymbol propertySymbol)
+                    continue;
+                if (semanticModel.GetSymbolInfo(property.Type).Symbol is not ITypeSymbol propertyTypeSymbol)
+                    continue;
+
+                if (IsSerializationIgnored(propertySymbol, bindAttribute))
```

Comment:
This only verifies that you used the bind attribute. But we should verify whether it is used with bind.none or verify that the type is serializable. Also, we should check JsonIgnoreAttributes